### DISCUSSION TITLE
How messages already delivered to consumer receive queue after consumer crashed.

### DISCUSSION THREAD

**Discussion by jaggerwang** [apache/pulsar#20780] [Q&A] [answered]
Does these messages will be consumed by other consumers? and how?

**Comment by tisonkun** [comment, marked as answer]
If the message isn't acked, and the consumer crashed is known by the broker, the broker will redeliver the message by [Routing modes](https://pulsar.apache.org/docs/3.0.x/concepts-messaging/#routing-modes).

**Reply by jaggerwang** [reply]
Does it means that the broker should save what messages have send to this crashed consumer?

**Reply by tisonkun** [reply]
No. Generally a broker knows what messages are acked and if one is not acked in within an interval, it resends. If the routing table changed, the message will be delivered to the new consumer based on the routing mode.

Specific Pulsar subscription is exclusive/failover which is only one serving, or shared which doesn't care which consumer consumes a specific message.
